### PR TITLE
We do actually want CI to pass a branch

### PR DESCRIPTION
**Pull request by DMRobertson**
this pr not from my fork but from the repo itself

### CODE REVIEW

**Review by richvdh — COMMENTED** [review]
looks sensible, but we should document it - https://github.com/matrix-org/sytest/blob/7178c8bd53557c739e99771bf67c9116ec0d731f/docker/README.md lists all the other env vars that the docker image understands, so we should include this one

File: docker/README.md [hunk 1]
```diff
@@ -17,9 +17,12 @@ Included currently is:
 
 ### Synapse
 
-The `sytest-synapse` images expect a checkout of the synapse git repository to
-be mounted at `/src`; additionally, server logs will be written to `/logs`, so
-it is useful to mount a volume there too.
+The `sytest-synapse` and `sytest-dendrite` images expect a checkout of the 
+homeserver git repository to be mounted at `/src`; otherwise, the image will
+try to fetch that repository from GitHub.
```
Comment (richvdh) [inline docker/README.md:22]: Is this correct? I thought that it would just fail if `/src` didn't exist.

File: docker/README.md [hunk 2]
```diff
@@ -29,6 +32,8 @@ docker run --rm -it -v /path/to/synapse\:/src:ro -v /path/to/where/you/want/logs
 
 The following environment variables can be set with `-e` to control the test run:
 
+ * `SYTEST_BRANCH`: controls which branch of synapse or dendrite we fetch when  `/src`
+   is missing. If omitted or if that branch doesn't exist, we'll fetch the  appropriate development branch.
```
Comment (richvdh) [inline docker/README.md:36]: this isn't quite right. As above: if `/src` is missing, I don't think it works at all...

**Review by richvdh — CHANGES_REQUESTED** [review]

File: docker/README.md (hunk 2 above)
Comment (DMRobertson) [reply docker/README.md:36]: Argh! Thanks. I think I keep confusing `/src` and `/sytest`. It's the CI job that has to fetch the source

Comment (richvdh) [reply docker/README.md]: No worries! It is pretty confusing.

File: docker/README.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 # SyTest Docker Images
 
 These Dockerfiles create containers for running SyTest in various
-configurations. SyTest is not included in these images, but its dependencies
-are.
+configurations. SyTest is not [included in these images by default](#selecting-a-sytest-checkout),
+but its dependencies are.
```
Comment (richvdh) [inline docker/README.md:5]: it's not actually possible to include Sytest itself in the Docker images. How about:


```suggestion
configurations. SyTest is not included in these images (see 
[Selecting a checkout of SyTest](#selecting-a-sytest-checkout)),
but its dependencies are.
```

File: docker/README.md [hunk 4]
```diff
@@ -68,11 +68,20 @@ it is useful to mount a volume there too.
...
+the same name as that of the target homeserver, or `develop`.
+This can be overridden with the environment variable `SYTEST_BRANCH`,
+for example:
+
+```
+docker run --rm -it \
+    -e SYTEST_BRANCH="my-sytest-branch"
+    -v /path/to/synapse\:/src:ro -v /path/to/where/you/want/logs\:/logs
+    matrixdotorg/sytest-synapse:buster
+```
```
Comment (richvdh) [inline docker/README.md:85]:
```suggestion

If the branch referred to by `SYTEST_BRANCH` does not exist, `develop` is used
instead.

```

**Review by richvdh — APPROVED** [review]
lgtm otherwise, thanks!

**Comment by DMRobertson** [comment]
I'll run the docker `build.sh` and `publish.sh` scripts so that synapse's CI can pick this up.

**Pull request merged**